src/engines/ioda/src/ioda/Engines/ObsStore/Type.cpp

```diff
@@ -81,6 +82,44 @@ bool Type::operator!=(const Type & rhs) const {
   return !(*this == rhs);
 }
 
+detail::Engines::HH::HH_Type Type::getHDF5Type() const {
+  // We can look at ObsTypes using getType() or type_ and return the matching HDF5 type.
+  using namespace detail::Engines::HH;
+  static auto HH_default_string_type
+    = std::dynamic_pointer_cast<HH_Type>(HH_Type_Provider::instance()->makeStringType(
+        typeid(char),Types::constants::_Variable_Length, StringCSet::UTF8).getBackend());
+
+  static const std::map<ObsTypes, HH_Type> mappings{
+    {ObsTypes::FLOAT, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(float))}},
+    {ObsTypes::DOUBLE, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(double))}},
+    {ObsTypes::LDOUBLE, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(long double))}},
+    {ObsTypes::SCHAR, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(signed char))}},
+    {ObsTypes::SHORT, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(short))}},
+    {ObsTypes::INT, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(int))}},
+    {ObsTypes::LONG, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(long))}},
+    {ObsTypes::LLONG, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(long long))}},
+    {ObsTypes::UCHAR, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(unsigned char))}},
+    {ObsTypes::UINT, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(unsigned int))}},
+    {ObsTypes::USHORT, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(unsigned short))}},
+    {ObsTypes::ULONG, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(unsigned long))}},
+    {ObsTypes::ULLONG, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(unsigned long long))}},
+    {ObsTypes::CHAR, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(char))}},
+    // No support for wchars, char16, char32. Unimplemented in the hdf5 backend's
+    // current iteration of the type system, and unused in ioda.
+    //{ObsTypes::WCHAR, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(wchar_t))}},
+    //{ObsTypes::CHAR16, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(char16_t))}},
+    //{ObsTypes::CHAR32, HH_Type{HH_Type_Provider::getFundamentalHHType(typeid(char32_t))}}
+    // No support for array types
+    //{ObsTypes::ARRAY, ...}
+    // We do support a single basic string type. ObsStore doesn't track the string's
+    // character set, length, or padding information, so we assume some defaults
+    // for the type conversion.
```

Comment:
I'm guessing this block of commented code is a placeholder for a future upgrade. If so, I'm good keeping it in.

Reply:
Yes, it's a placeholder for any further changes to the type system.

Reply:
Thanks for the explanation. I think it's okay to keep this in, no need to change this.